holoviews/core/data/grid.py

```diff
@@ -115,6 +117,31 @@ def init(cls, eltype, data, kdims, vdims):
                             'actual shape: %s' % (vdim, valid_shape, shape), cls)
         return data, {'kdims':kdims, 'vdims':vdims}, {}
 
+    @classmethod
+    def concat(cls, datasets, dimensions, vdims):
+        from . import Dataset
+        with sorted_context(False):
+            datasets = NdMapping(datasets, kdims=dimensions)
+            datasets = datasets.clone([(k, v.data if isinstance(v, Dataset) else v)
+                                       for k, v in datasets.data.items()])
+        if len(datasets.kdims) > 1:
+            items = datasets.groupby(datasets.kdims[:-1]).data.items()
+            return cls.concat([(k, cls.concat(v, v.kdims, vdims=vdims)) for k, v in items],
+                              datasets.kdims[:-1], vdims)
+        return cls.concat_dim(datasets, datasets.kdims[0], vdims)
+
+
+    @classmethod
+    def concat_dim(cls, datasets, dim, vdims):
+        values, grids = zip(*datasets.items())
+        new_data = {k: v for k, v in grids[0].items() if k not in vdims}
+        new_data[dim.name] = np.array(values)
+        for vdim in vdims:
+            arrays = [grid[vdim.name] for grid in grids]
+            stack = np.stack if any(is_dask(arr) for arr in arrays) else da.stack
+            new_data[vdim.name] = stack(arrays, -1)
+        return new_data
```

Comment:
Since arrays cannot be concatenated along multiple axes at once the implementation of ``concat`` on gridded interfaces has two components. A general ``concat`` method coordinates hierarchical concatenation along each dimension and uses the interface specific ``concat_dim`` method implementations to concatenate along one particular axis or dimension.